static/js/hotkey.js

```diff
@@ -166,7 +166,9 @@ exports.process_escape_key = function (e) {
     }
 
     if (message_edit.is_editing(current_msg_list.selected_id())) {
-        message_edit.end(current_msg_list.selected_row());
+        row = $(".message_edit_content").filter(":focus").closest(".message_row");
```

Comment:
why not just `row = current_msg_list.selected_row();`?  That feels cleaner.

Reply:
It's partly for consistency, as every other call to message_edit.end() provides this definition of "row" as its argument, including the click event for closing the message edit box (see click_handlers.js:237). 

It's also partly because `current_msg_list.selected_row()` provides a different object than `$(".message_edit_content").filter(":focus").closest(".message_row")`. The former provides an object containing only the message row, while the latter provides that object and a prevObject referencing the edit form that was cancelled. 
This is the ![former](https://user-images.githubusercontent.com/12477149/31798991-56ca3d1c-b505-11e7-9b80-046f62e13f22.png)  and 
![this](https://user-images.githubusercontent.com/12477149/31798998-6a713a78-b505-11e7-9b0a-45d82118d2bc.png) is the latter.

That change is what correctly sets the focus.

Reply:
Hmm.  We should at the very least add a comment here, because the `message_edit.is_editing(current_msg_list.selected_id())` line really makes it look like one should be interacting with `current_msg_list.selected_row()` on the next line (that's usually how we interact with the selected row), and I could easily see someone "cleaning this up" and accidentally breaking it.